yt_dlp/extractor/asobistage.py

```diff
@@ -0,0 +1,160 @@
+        event_title = traverse_obj(event_data, ('props', 'pageProps', 'eventCMSData', 'event_name', {str}))
+        if not all((event_id, event_slug, event_title)):
+            raise ExtractorError('Unable to get required event data')
+
+        channel_list_url = functools.reduce(urljoin, [
+            self._search_regex(
+                r'"(?P<url>https://asobistage\.asobistore\.jp/cdn/[^/]+/)', webpage,
+                'cdn endpoint url', group=('url')),
+            'events/', f'{event_id}/', f'{video_type_id}.json'])
+        channels_json = self._download_json(
+            channel_list_url, video_id, fatal=False,
+            note='Getting channel list', errnote='Unable to get channel list')
+        available_channels = traverse_obj(channels_json, (
+            video_type_id, lambda _, v: v['broadcast_slug'] == event_slug, 'channels',
+            lambda _, v: v['chennel_vspf_id'] != '00000', {dict}))
+
+        owned_tickets = set(self._get_owned_tickets(video_id))
+        if not owned_tickets.intersection(traverse_obj(
+                available_channels, (..., 'viewrights', ..., 'tickets', ..., 'digital_product_id', {str_or_none}))):
+            raise ExtractorError('No valid ticket for this event', expected=True)
+
+        channel_ids = traverse_obj(available_channels, (..., 'chennel_vspf_id', {str}))
+
+        token = self._get_token(video_id)
+
+        entries = []
+        for channel_id in channel_ids:
+            channel_data = {}
+
+            if video_type_name == 'archive':
+                channel_json = self._download_json(
+                    f'https://survapi.channel.or.jp/proxy/v1/contents/{channel_id}/get_by_cuid', f'{video_id}/{channel_id}',
+                    note='Getting archive channel info', errnote='Unable to get archive channel info',
+                    headers={'Authorization': f'Bearer {token}'})
+                channel_data = traverse_obj(channel_json, ('ex_content', {
+                    'm3u8_url': 'streaming_url',
+                    'title': 'title',
+                    'thumbnail': ('thumbnail', 'url'),
+                }))
+            elif video_type_name == 'player':
+                channel_json = self._download_json(
+                    f'https://survapi.channel.or.jp/ex/events/{channel_id}', f'{video_id}/{channel_id}',
+                    note='Getting live channel info', errnote='Unable to get live channel info',
+                    headers={'Authorization': f'Bearer {token}'}, query={'embed': 'channel'})
+                channel_data = traverse_obj(channel_json, ('data', {
+                    'm3u8_url': ('Channel', 'Custom_live_url'),
+                    'title': 'Name',
+                    'thumbnail': 'Poster_url',
+                }))
+
+            m3u8_url = channel_data.get('m3u8_url')
+            if not m3u8_url:
+                raise ExtractorError('Unable to get channel m3u8 url')
```

Comment:
Maybe just `self.report_warning` and `continue` if no m3u8 URL was found since we are returning a playlist?